src/components/permissible.js

```diff
@@ -20,7 +20,9 @@ export function Permissible(
       history: PropTypes.object, // eslint-disable-line react/forbid-prop-types
     };
 
-    componentWillMount() {
+    constructor(props) {
```

Comment:
can you elaborate a little bit why you changed `componentWillMount` to `constructor`? Wouldn't be `componentDidMount` more appropriate? What do you think?

Reply:
oh right, I see what you mean. Okay then

Reply:
Just to clarify, a component will run method `componentDidMount` even if render returns just `null`. Also fact that `constructor` has been called doesn't mean that it was rendered.

Sandbox:
https://codesandbox.io/s/0oo4583qr0

Reply:
Thanks for the input @szymonbrainhub